examples/master/freedom-k64f/freedom_k64f_accel_map_example.cc

```diff
@@ -0,0 +1,265 @@
+#include "kickcat/Prints.h"
+#include "kickcat/helpers.h"
+
+using namespace kickcat;
+
+// PDO mapping configuration
+namespace pdo
+{
+    // TxPDO mapping (slave -> master): Accelerometer only
+    constexpr uint32_t tx_mapping[] = {
+        0x60000010, // accel_x, 16 bits
+        0x60010010, // accel_y, 16 bits
+        0x60020010  // accel_z, 16 bits
+    };
+    constexpr uint32_t tx_mapping_count = sizeof(tx_mapping) / sizeof(tx_mapping[0]);
+
+    // RxPDO mapping (master -> slave): RGB LEDs
+    constexpr uint32_t rx_mapping[] = {
+        0x70000008, // LED_R, 8 bits
+        0x70010008, // LED_G, 8 bits
+        0x70020008  // LED_B, 8 bits
+    };
+    constexpr uint32_t rx_mapping_count = sizeof(rx_mapping) / sizeof(rx_mapping[0]);
+}
+
+namespace freedom
+{
+    // POC 1: Only accelerometer data
+    struct fxos8700cq_accel_only
+    {
+        int16_t accelerometerX; // mapped 0x6000
+        int16_t accelerometerY; // mapped 0x6001
+        int16_t accelerometerZ; // mapped 0x6002
+    } __attribute__((packed));
+
+    struct Input
+    {
+        fxos8700cq_accel_only sensor;
+    } __attribute__((packed));
+
+    struct Output
+    {
+        uint8_t LED_R; // mapped 0x7000
+        uint8_t LED_G; // mapped 0x7001
+        uint8_t LED_B; // mapped 0x7002
+    } __attribute__((packed));
+}
+
+int main(int argc, char *argv[])
+{
+    if (argc != 3 and argc != 2)
+    {
+        printf("usage redundancy mode : ./test NIC_nominal NIC_redundancy\n");
+        printf("usage no redundancy mode : ./test NIC_nominal\n");
+        return 1;
+    }
+
+    std::string red_interface_name = "";
+    std::string nom_interface_name = argv[1];
+    if (argc == 3)
+    {
+        red_interface_name = argv[2];
+    }
+
+    std::shared_ptr<AbstractSocket> socket_nominal;
+    std::shared_ptr<AbstractSocket> socket_redundancy;
+    try
+    {
+        auto [nominal, redundancy] = createSockets(nom_interface_name, red_interface_name);
+        socket_nominal = nominal;
+        socket_redundancy = redundancy;
+    }
+    catch (std::exception const &e)
+    {
+        std::cerr << e.what() << std::endl;
+        return 1;
+    }
+
+    auto report_redundancy = []()
+    {
+        printf("Redundancy has been activated due to loss of a cable \n");
+    };
+
+    std::shared_ptr<Link> link = std::make_shared<Link>(socket_nominal, socket_redundancy, report_redundancy);
+    link->setTimeout(2ms);
+    link->checkRedundancyNeeded();
+
+    Bus bus(link);
+
+    auto print_current_state = [&]()
+    {
+        for (auto &slave : bus.slaves())
+        {
+            State state = bus.getCurrentState(slave);
+            printf("Slave %d state is %s\n", slave.address, toString(state));
+        }
+    };
+
+    uint8_t io_buffer[2048];
+
+    // Copied from Ingenia Control example
+    const auto mapPDO = [&](const uint8_t slaveId, const uint16_t PDO_map, const uint32_t *data, const uint32_t dataSize, const uint32_t SM_map) -> void
```

Comment:
Should we make this a global method that can be used by user to remap/reconfigure?

Reply:
Seems to be a good idea to create a generic helper. However we need to add an emulated version of complete access to do so: ingenia is quite the exception to support this option.
I already wrote an emulated compelte access for read that can be serve as an example, and the API is ready (better: read and write will be similar).